src/main/java/core/basesyntax/service/impl/operation/SupplyOperation.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.impl.operation;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.model.Fruit;
+
+public class SupplyOperation implements OperationHandler {
+    private final FruitStorageDao fruitStorageDao;
+
+    public SupplyOperation(FruitStorageDao fruitStorageDao) {
+        this.fruitStorageDao = fruitStorageDao;
+    }
+
+    @Override
+    public boolean doOperation(Fruit fruit, int quantity) {
+        if (fruit == null) {
+            throw new RuntimeException("Fruit can't be null");
+        }
+        if (quantity < MIN_FRUIT_QUANTITY) {
```

Comment:
It seems like `MIN_FRUIT_QUANTITY` is a constant that should be defined in this class, but it's missing. Make sure to define this constant properly if it's intended to be used here, or use another way to check for valid quantity values.